extensions/resteasy-reactive/quarkus-resteasy-reactive-qute/runtime/src/main/java/io/quarkus/resteasy/reactive/qute/runtime/TemplateResponseUniHandler.java

```diff
@@ -0,0 +1,41 @@
+package io.quarkus.resteasy.reactive.qute.runtime;
+
+import static io.quarkus.resteasy.reactive.qute.runtime.Util.*;
+import static io.quarkus.resteasy.reactive.qute.runtime.Util.toUni;
+
+import org.jboss.resteasy.reactive.server.core.ResteasyReactiveRequestContext;
+import org.jboss.resteasy.reactive.server.spi.ServerRestHandler;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.qute.Engine;
+import io.quarkus.qute.TemplateInstance;
+import io.smallrye.mutiny.Uni;
+
+public class TemplateResponseUniHandler implements ServerRestHandler {
+
+    private volatile Engine engine;
+
+    @Override
+    public void handle(ResteasyReactiveRequestContext requestContext) {
+        Object result = requestContext.getResult();
+        if (!(result instanceof TemplateInstance)) {
+            return;
+        }
+
+        if (engine == null) {
```

Comment:
This is not the first time I can see a volatile access like this in a `ServerRestHandler`. Would it make sense to make it possible to initialize a `ServerRestHandler` instance so that you can avoid this lazy init?

Reply:
Not easily unfortunately